src/Utilities/cachingoptimizer.jl

```diff
@@ -148,16 +148,57 @@ function attach_optimizer(model::CachingOptimizer)
     # We do not need to copy names because name-related operations are handled by `m.model_cache`
     indexmap = MOI.copy_to(model.optimizer, model.model_cache, copy_names=false)
     model.state = ATTACHED_OPTIMIZER
-    # MOI does not define the type of index_map, so we have to copy it into a
-    # concrete container. Also load the reverse map.
-    model.model_to_optimizer_map = IndexMap()
-    model.optimizer_to_model_map = IndexMap()
-    for k in keys(indexmap)
-        model.model_to_optimizer_map[k] = indexmap[k]
-        model.optimizer_to_model_map[indexmap[k]] = k
+    # MOI does not define the type of index_map, so we might have to copy it
+    # into a concrete container. Also load the reverse map.
+    model.model_to_optimizer_map = convert(IndexMap, indexmap)#IndexMap()
+    model.optimizer_to_model_map = reverse_index_map(indexmap)#IndexMap()
+    nothing
+end
+
+function reverse_index_map(src::IndexMap)
+    dest = IndexMap()
+    sizehint!(dest.varmap, length(src.varmap))
+    # sizehint!(dest.conmap, length(src.conmap))
+    reverse_dict(dest.varmap, src.varmap)
+    reverse_dict(dest.conmap, src.conmap)
+    return dest
+end
+function reverse_dict(dest::AbstractDict, src::AbstractDict)
+    for (k,v) in src
+        dest[v] = k
     end
 end
 
+function Base.convert(::Type{IndexMap}, d::AbstractDict{MOI.Index, MOI.Index})
+    map = IndexMap()
+    for (k,v) in d
+        map[k] = v
+    end
+    return map
+end
+function Base.convert(::Type{IndexMap}, d::IndexMap)
+    # return d
+    # if we return d as is, its not possible to add variables
+    # if there was a dense dict inside
+    # the solution would be to allow automatically swtiching
+    # a ClevelDenseDict...
```

Comment:
I don't understand this comment or this function. What does it mean to convert an `IndexMap` to an `IndexMap`?

Reply:
the varmap field might be a standard dict or a dense dict. dense dict cannot grow, they are fixed.
One possible smooth solution for that is doing something similar from CleverDict and automating the switching between the inner dict types.

Reply:
`_standardize(::IndexMap)`